### PR TITLE
rewrite(server): Rewrite server to TypeScript

### LINKED ISSUE #10: todo(server): rewrite in TS
going to actually be taking this one myself and doing it ASAP

### PR DESCRIPTION
**Pull request by AceLikesGhosts**
Beginning stages for #10 and #11 

#11 will be completed once this PR is merged, as then we will have an easily extendable Server setup.

What this PR does:
1. Changes from a single file server to a modular server setup, so that we are in the future able to have both an RPC-express server and a BetterDiscord plugin which will communicate with WebSockets
2. Rewrites the project in Typescript, adding type safety and making the project not so.. ugly
3. Hopefully will bring more clarity as to what does what, with more distinct files, and not having everything in a singular file
4. Add linting to the file (`@biased-ts/eslint-config`)

### CODE REVIEW

**Comment by l1mey112** [comment]
hi honey

**Comment by AceLikesGhosts** [comment]
> hi honey

hi, source code review when? im rusty @ TS i gotta make sure im doing all the things the cool kids do for it (i even threw in eslint!! :o)

File: src/utils/index.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        };
+    }
+    else {
+        return {
+            details: `Paused: ${song}`,
+            state: artist,
+            largeImageKey: icon || Globals.images.default_img,
+            largeImageText: song,
+            smallImageKey: Globals.images.pause_img || undefined,
+            smallImageText: Globals.images.pause_img !== undefined ? 'Paused' : undefined,
```
Comment (AceLikesGhosts) [inline src/utils/index.ts]: Why are we not just passing `Globals` as we are to the generic `Server`

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+export const Globals: IGlobals = {
+    client_id: process.env.CLIENT_ID || '1075993095138713612',
+    port: Number(process.env.PORT) || 2134,
+    images: {
+        default_img: process.env.DEFAULT_IMG || 'ytm',
+        pause_img: process.env.PAUSE_IMG || 'paused',
+        play_img: process.env.PLAY_IMG || 'playing'
+    }
+};
```
Comment (AceLikesGhosts) [inline src/index.ts]: Stop exporting this, rename it, and pass it along.

**Review by AceLikesGhosts — COMMENTED** [review]
cause not on pc

File: src/index.ts (hunk 2 above)
Comment (AceLikesGhosts) [reply src/index.ts]: dep injection :D

File: src/utils/index.ts (hunk 1 above)
Comment (AceLikesGhosts) [reply src/utils/index.ts]: rewrite this to take 2 objects, song data, then constants

File: src/index.ts (hunk 2 above)
Comment (AceLikesGhosts) [reply src/index.ts]: commit 844760badf5eb1edcbe2182fa5ac1f995732d9bf resolves this

File: src/utils/index.ts (hunk 1 above)
Comment (AceLikesGhosts) [reply src/utils/index.ts]: commit 844760badf5eb1edcbe2182fa5ac1f995732d9bf resolves this

**Comment by AceLikesGhosts** [comment]
This is ready to merge but I'm going to wait for LM's review (: 
image

**Review by l1mey112 — APPROVED** [review]
i approve this. nice work ace

**Comment by AceLikesGhosts** [comment]
bro basically "LGTM"'d with no actual review notes 😭

**Comment by l1mey112** [comment]
> bro basically "LGTM"'d with no actual review notes 😭

because it does look good to me

**Comment by AceLikesGhosts** [comment]
> > bro basically "LGTM"'d with no actual review notes 😭
> 
> because it does look good to me

this project is `void` abuse

**Pull request merged**